dashboard/lib/policies/child_account.rb

```diff
@@ -94,6 +94,14 @@ def self.lockout_date(user)
     state_policy(user).try(:[], :lockout_date)
   end
 
+  # Checks if the user is locked out due to non-compliance with CAP.
+  def self.locked_out?(user)
+    user_lockout_date = lockout_date(user)
+    return false unless user_lockout_date
+
+    DateTime.now >= user_lockout_date || !user_predates_policy?(user)
```

Comment:
This is the same as:

```
# Not locked out if the user account predates the policy and is in their grace period
return false if user_predates_policy?(user) && DateTime.now < user_lockout_date

# Otherwise, the account is subject to the policy, so we assume the account is locked out
return true
```

Where it might be easier to follow the logic if it is a list of exceptions since you have a `return false` earlier and the exceptions are the important parts to understand on a future reading.

The logic does hold though. It looks good overall.

Reply:
@dayne @wilkie I believe it makes sense to check `user_predates_policy?` (users created before July 1, 2023) only before July 1, 2024, since all users created after that date should be locked out anyway based on legal requirements:
https://github.com/code-dot-org/code-dot-org/blob/55b1a5afdc90eb4606535921df91a8eb2291cfe5/dashboard/lib/policies/child_account.rb#L79-L89

Reply:
@daynew Shouldn't we also update the method `user_predates_policy?` to check users created before `July 1, 2024` instead of `July 1, 2023` now?

Reply:
I think we should leave the `user_predates_policy?` logic for now. I think we will end up deprecating this function once we implement the "Grace Period" feature.